utils.go

```diff
@@ -54,3 +55,46 @@ func validateIQN(iqn string) error {
 	}
 	return nil
 }
+
+func filterIPsForInterface(ifaceName string, ipAddress ...string) ([]string, error) {
+	filterredIPs := make([]string, 0)
+	iface, err := net.InterfaceByName(ifaceName)
+	if err != nil {
+		fmt.Printf("\nError could not find interface %s : %v", ifaceName, err)
+		return filterredIPs, err
+	}
+
+	addrs, err := iface.Addrs()
+	if err != nil {
+		fmt.Printf("\nError failed to get addresses of interface %s : %v", ifaceName, err)
+		return filterredIPs, err
+	}
+
+	for _, ipAddr := range ipAddress {
+
+		ip := net.ParseIP(ipAddr)
+		if ip == nil {
+			fmt.Printf("\nError invalid IP address: %s", ipAddr)
+			continue
+		}
+		found := false
+		for _, addr := range addrs {
+			ifaceIP, ifaceSubnet, err := net.ParseCIDR(addr.String())
+			if err != nil {
+				fmt.Printf("\nError failed to parse address %s of interface %s : %v", addr.String(), ifaceName, err)
+				continue
+			}
+
+			// Check if the IP belongs to the subnet
+			if ifaceSubnet.Contains(ip) || ifaceIP.Equal(ip) {
+				found = true
```

Comment:
Should we break the for loop here?

Reply:
Thanks for the feedback! I've added the break statement.